src/main/webapp/resources/js/pages/projects/share/ShareMetadata.jsx

```diff
@@ -0,0 +1,148 @@
+} from "../../../apis/metadata/field";
+import {
+  compareRestrictionLevels,
+  getColourForRestriction,
+} from "../../../utilities/restriction-utilities";
+import { setMetadataRestrictions } from "./shareSlice";
+import { TargetMetadataRestriction } from "./TargetMetadataRestriction";
+
+/**
+ * React component to display metadata restrictions.
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export function ShareMetadata() {
+  const dispatch = useDispatch();
+  /**
+   * Available restrictions for metadata fields
+   */
+  const [restrictions, setRestrictions] = React.useState([]);
+
+  const { currentProject, projectId, metadataRestrictions } = useSelector(
+    (state) => state.shareReducer
+  );
+
+  /**
+   * Get the fields for the current project.  The restrictions from these fields
+   * will act as a base for the restriction level when the fields are shared.
+   */
+  const { data: sourceFields } = useGetMetadataFieldsForProjectQuery(
+    currentProject
+  );
+
+  /**
+   * Target project metadata fields. Needed to determine, which fields will be
+   * on both projects so that the new restriction for a field should start at
+   * the highest level of restriction.
+   */
+  const { data: targetExistingFields } = useGetMetadataFieldsForProjectQuery(
+    projectId,
+    {
+      skip: !projectId,
+    }
+  );
+
+  React.useEffect(() => {
+    if (sourceFields && targetExistingFields) {
+      const existing = targetExistingFields.reduce(
+        (fields, field) => ({
+          ...fields,
+          [field.fieldKey]: field,
+        }),
+        {}
+      );
+      const fields = sourceFields.map((field) => {
+        const f = { ...field, current: field.restriction };
+        delete f.initial;
+        if (existing[f.fieldKey]) {
+          /*
+          If field exists in target project
+           */
+          f.target = existing[f.fieldKey].restriction;
+          f.difference = compareRestrictionLevels(f.restriction, f.target);
+          f.restriction = f.difference < 0 ? f.target : f.current;
+        }
+        return f;
+      });
+      dispatch(setMetadataRestrictions(fields));
+    } else if (sourceFields) {
+      // Allow user to see what the restrictions are on the source fields
+      dispatch(setMetadataRestrictions(sourceFields));
+    }
+  }, [dispatch, sourceFields, targetExistingFields]);
+
+  /**
+   * On load, get metadata restrictions that are possible for a project.
+   * These are formatted for Select inputs ({label, value}).
+   */
+  React.useEffect(() => {
+    getMetadataRestrictions().then(setRestrictions);
+  }, []);
+
+  const columns = [
+    {
+      title: i18n("ShareMetadata.field"),
+      key: "label",
+      dataIndex: "label",
+    },
+    {
+      title: i18n("ShareMetadata.current"),
+      key: "current",
+      dataIndex: "current",
+      render(text, item, index) {
+        const field = restrictions.find(
+          (r) => r.value === sourceFields[index].restriction
```

Comment:
Change variable `r` to something more descriptive

Reply:
Updated in 23d13a67117635b888bea9344f35f10032628d41